src/components/hero.js

```diff
@@ -1,114 +1,251 @@
-import React from "react"
+import React, { useEffect, useRef, useState } from "react"
 import { Box, Text, Button, Flex } from "rebass"
-import { heroImg, heroBrandmark } from "../images"
+import { fabric } from "fabric"
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome"
+import { faFacebookSquare } from "@fortawesome/free-brands-svg-icons"
+
+import theme from "../theme"
+import { heroImg, brandmarkPath } from "../images"
 import Container from "./container"
 
-const Hero = () => (
-  <Box
-    sx={{
-      position: "relative",
-      backgroundImage: `url("${heroImg}")`,
-      backgroundRepeat: "no-repeat",
-      backgroundPosition: "center",
-      backgroundSize: "cover",
-      width: "100%",
-      height: "100vh",
-      mt: "-105px",
-      mb: 4,
-    }}
-  >
+const getScaledHeroImage = (canvas, imageEl) => {
+  const heroImage = new fabric.Image(imageEl)
+  const canvasRatio = canvas.width / canvas.height
+  const imageRatio = imageEl.width / imageEl.height
+
+  if (imageRatio > canvasRatio) {
+    heroImage.scaleToHeight(canvas.height)
+  } else {
+    heroImage.scaleToWidth(canvas.width)
+  }
+
+  return heroImage
+}
+
+const rotatePath = (path, midpoint, { point, angle }) => {
+  const rotatePoint = fabric.util.rotatePoint(
+    point,
+    new fabric.Point(midpoint.x + point.x, midpoint.y + point.y),
+    fabric.util.degreesToRadians(angle)
+  )
+  path.set({
+    top: rotatePoint.y,
+    left: rotatePoint.x,
+    angle,
+  })
+}
+
+const drawRotatingBrandmark = (canvas, heroImageTemplate) => {
+  const heroImage = fabric.util.object.clone(heroImageTemplate)
+
+  const clipPath = new fabric.Path(brandmarkPath)
+  clipPath.absolutePositioned = true
+  clipPath.scale((1.2 * canvas.height) / clipPath.height)
+
+  const ANIMATION_TIME = 60 * 1000
+  const INITIAL_ANGLE = -15
+  const clipPathMidpoint = new fabric.Point(
+    500 * clipPath.getObjectScaling().scaleX,
+    581.9 * clipPath.getObjectScaling().scaleY
+  )
+  let startTimestamp = 0
+  let angle = INITIAL_ANGLE
+
+  const drawClip = timestamp => {
+    if (window.innerWidth === canvas.width) {
```

Comment:
potentially dangerous that may introduce weird behavior. 
I assume you are using `useEffect` for this from below. Any layout specific behavior such as measurements should be done before browser repaint. Might cause jitters, not sure. Only a theory.

Reply:
This is being called independently of `useEffect` and [is guaranteed to run before a repaint](https://developer.mozilla.org/en-US/docs/Web/API/window/requestAnimationFrame).

Reply:
useEffect itself asynchronously run after browser paint. so any code inside that effect will only be scheduled so. So your animation although will guarantee to run before a repaint, it will only be called when useEffect is next on the event loop (the updated view was committed already though). This is the possible different "lifecycle" I was pertaining to. Though I'm not 100% sure, as long as it doesn't do any weird behavior, just taking into mind the jitters I saw back then which I think was caused by a different problem making this concern invalid, I'm fine with it. =)) Let's resolve this.